pandas/tests/frame/test_ufunc.py

```diff
@@ -117,6 +117,27 @@ def test_binary_frame_series_raises():
         np.logaddexp(df["A"], df)
 
 
+def test_unary_accumulate_axis():
+    # https://github.com/pandas-dev/pandas/issues/39259
+    df = pd.DataFrame({"a": [1, 3, 2, 4]})
+    result = np.maximum.accumulate(df)
+    expected = pd.DataFrame({"a": [1, 3, 3, 4]})
+    tm.assert_frame_equal(result, expected)
+
+    df = pd.DataFrame({"a": [1, 3, 2, 4], "b": [0.1, 4.0, 3.0, 2.0]})
+    result = np.maximum.accumulate(df)
+    # in theory could preserve int dtype for default axis=0
+    expected = pd.DataFrame({"a": [1.0, 3.0, 3.0, 4.0], "b": [0.1, 4.0, 4.0, 4.0]})
+    tm.assert_frame_equal(result, expected)
```

Comment:
Mmm, I'd like this to be more than "in theory". I'd consider this a buggy test, since things should be done blockwise for axis=0.

Can you you change the test case to have just floats or just ints (even if you have to manually split it for test coverage?).

Reply:
I specifically used two dtypes to have two blocks to ensure we handle this case correctly for axis=1 (which can never be done clockwise)

Just above there is already a case with only ints that preserves the int dtype.

Reply:
Note that before pandas 1.2.0, this also didn't preserve the dtypes per column, and 1.2.0 itself didn't calculate a proper result (so I would call this PR a strict improvement ;))

Reply:
agreed re preserving for axis=0.  couldn't we still use mgr.apply in that case?

Reply:
To be clear, I *could* certainly detect the special case of accumulate with axis=0, and then pass axis=1 to the blocks, but:

1\) that requires special case code like this in array_ufunc:

```python
else:
    # the ufunc(dataframe) case
    ...
    elif method == "accumulate" and ("axis" not in kwargs or ("axis" in kwargs and kwargs["axis"] == 0)):
        # swap axis for the transposed Block values
        kwargs["axis"] = 1
        result = mgr.apply(getattr(ufunc, method), **kwargs)
```

2\) that requires `Block.apply` to be "aware" of `axis`. Currently it simply passes through keywords, but in this case it would need to interpret `axis` differently depending on whether its values are stored as 2D or 1D (and I know we already need to take this axis swapping into account in many places, eg with `NDFrame._get_block_manager_axis(axis)`, and in the internals as well, but that's typically when axis is a keyword of our own, and not a user-specified kwarg of a generic applied function). So I certainly could add an ExtensionBlock.apply override to take this into account. But we could also decide to leave this as is for now.

Reply:
> but in this case it would need to interpret axis differently depending on whether its values are stored as 2D or 1D 

Once more with feeling: this wouldn't be an issue with 2D EAs.

Reply:
> Once more with feeling: this wouldn't be an issue with 2D EAs.

And I can also say: this wouldn't be an issue with only 1D arrays ..

Reply:
@TomAugspurger @jbrockmendel I opened https://github.com/pandas-dev/pandas/issues/39275 to keep track of the fact that this can be improved to preserve dtypes